packages/base/source/adt/union/union.js

```diff
@@ -82,8 +108,11 @@ instead to check if a value belongs to the ADT variant.`);
        *   where 'b = 'a[`@@folktale:adt:tag]
        */
       matchWith(pattern) {
-        return pattern[name](this);
-      } 
+        assertObject('Union.mapObject#matchWith', pattern);
+        if (name in pattern) { return pattern[name](this); }
+        else if (ANY in pattern) { return pattern[ANY](this) }
```

Comment:
This is sad, though, because I envisioned it being like Exlixir's case/cond matchers. It would just forward the results along. I guess with Union it doesn't really make sense, I'm not experienced enough to know. Any ideas how to optionally include the payload would be cool.